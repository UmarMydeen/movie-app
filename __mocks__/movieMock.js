const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster',
    imdbRating: '9.2',
    Poster: 'mockUrl'
}

export default movieMock;