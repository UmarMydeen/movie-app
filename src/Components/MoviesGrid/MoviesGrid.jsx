import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import fetchMovieList from '../../actions/fetchMovieList';
import MovieCard from '../MovieCard/MovieCard';
import {
    getMovieList,
    getMovieListDataSource,
} from '../../Reducers/index'

const MoviesGrid = ({
    movieList,
    dataSource
}) => {
    return (
        <div className="row row-cols-6 row-cols-md-5">
            {
                dataSource.length === 0 && (<p> No Data Found</p>)
            }
            {
                dataSource && dataSource.map(movie => (<MovieCard movie={movie} />))
            }
        </div>
    );
}

MoviesGrid.propTypes = {
    movieList: PropTypes.array.isRequired,
    dataSource: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
    return {
        movieList: getMovieList(state),
        dataSource: getMovieListDataSource(state),
    }
};

const hocChain = compose(
    connect(mapStateToProps, null),
);

export default hocChain(MoviesGrid);