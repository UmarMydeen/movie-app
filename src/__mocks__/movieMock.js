const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster'
}

export default movieMock;