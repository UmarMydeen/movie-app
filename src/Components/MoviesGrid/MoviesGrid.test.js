import { render, screen } from '@testing-library/react';
import { TestableMoviesGrid as MoviesGrid } from './MoviesGrid';
import movieMock from '../../__mocks__/movieMock';

let mockProps;

beforeEach(() => {
  mockProps = {
    movieList: [movieMock],
    dataSource: [movieMock]
  }
})

test('should render component correctly', () => {
  const component = render(<MoviesGrid {...mockProps} />);
  expect(component).toMatchSnapshot();
});

test('should render no data found when movie list is not present', () => {
  mockProps.movieList = [];
  mockProps.dataSource = [];
  const component = render(<MoviesGrid {...mockProps} />);
  expect(component).toMatchSnapshot();
});