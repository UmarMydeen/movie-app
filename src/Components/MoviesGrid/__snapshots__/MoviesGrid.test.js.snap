// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row row-cols-6 row-cols-md-5"
      >
        <a
          href="/movieDetails/mockimdbID1"
        >
          <div
            class="card m-1"
            data-testid="mockimdbID1"
          >
            <img
              alt="..."
              class="card-img-top"
              src="mockUrl"
            />
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                mockTitle
              </h5>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row row-cols-6 row-cols-md-5"
    >
      <a
        href="/movieDetails/mockimdbID1"
      >
        <div
          class="card m-1"
          data-testid="mockimdbID1"
        >
          <img
            alt="..."
            class="card-img-top"
            src="mockUrl"
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              mockTitle
            </h5>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render no data found when movie list is not present 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row row-cols-6 row-cols-md-5"
      >
        <p>
           No Data Found
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row row-cols-6 row-cols-md-5"
    >
      <p>
         No Data Found
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
